cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -507,6 +507,41 @@ TEST_F(AzuriteFileSystemTest, CreateDirUri) {
   ASSERT_RAISES(Invalid, fs_->CreateDir("abfs://" + RandomContainerName(), true));
 }
 
+TEST_F(AzuriteFileSystemTest, DeleteContainerDirSuccess) {
+  auto container_name = RandomContainerName();
+  ASSERT_OK(fs_->CreateDir(container_name));
+  ASSERT_OK(fs_->DeleteDir(container_name));
+}
+
+TEST_F(AzuriteFileSystemTest, DeleteDirSuccess) {
```

Comment:
I think it would be good to add a test case for a non-empty "virtual directory" where we expect the contents of the directory to be deleted. 

Would be nice to add a non-empty test case for hierarchical namespace too.

Reply:
> I think it would be good to add a test case for a non-empty "virtual directory" where we expect the contents of the directory to be deleted.

OK. I'll add it. 

> Would be nice to add a non-empty test case for hierarchical namespace too.

`AzureHierarchicalNamespaceFileSystemTest.DeleteDirSuccess` is for the case.